arangodb-net-standard/IndexApi/IndexApiClient.cs

```diff
@@ -0,0 +1,172 @@
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<GetIndexResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Delete an index permanently.
+        /// </summary>
+        /// <param name="indexId">The identifier of the index.</param>
+        /// <returns></returns>
+        public virtual async Task<DeleteIndexResponse> DeleteIndexAsync(string indexId)
+        {
+            string uri = _indexApiPath + "/" +indexId;
+            using (var response = await _client.DeleteAsync(uri).ConfigureAwait(false))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<DeleteIndexResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+
+        /// <summary>
+        /// Fetch the list of indexes for a collection.
+        /// </summary>
+        /// <param name="query">Query parameters for the request.</param>
+        /// <returns></returns>
+        public virtual async Task<GetAllCollectionIndexesResponse> GetAllCollectionIndexesAsync(GetAllCollectionIndexesQuery query)
+        {
+            string uri = _indexApiPath;
+            if (query == null)
+                throw new System.Exception("query is required");
+            if (string.IsNullOrEmpty(query.CollectionName))
+                throw new System.Exception("Collection name is required");
+
+            uri += '?' + query.ToQueryString();
+            using (var response = await _client.GetAsync(uri).ConfigureAwait(false))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<GetAllCollectionIndexesResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Creates an index
+        /// </summary>
+        /// <param name="indexType">Type of index to create</param>
+        /// <param name="query">Querystring details</param>
+        /// <param name="body">CreateIndexBody object containing the index details</param>
+        /// <returns></returns>
+        public virtual async Task<IndexResponseBase> PostIndexAsync(IndexType indexType, PostIndexQuery query, PostIndexBody body)
+        {
+            string uri = _indexApiPath;
+            if (query == null)
+                throw new System.Exception("query is required");
```

Comment:
I think we should use `ArgumentException` instead of `Exception` for all the validation exceptions. Using a more specialized exception type will be clearer for callers when debugging or handling exceptions.

Example:
```c#
if (query == null)
    throw new System.ArgumentException("query is required", nameof(query));
if (string.IsNullOrEmpty(query.CollectionName))
    throw new System.ArgumentException("Collection name is required", nameof(query));
if (body == null)
    throw new System.ArgumentException("body is required", nameof(body));
```

Also consider documenting the thrown exception in the XML doc for both the interface and the implementation. I think it would make sense to make this a requirement of interface contract.

Example:
```c#
/// <summary>
/// Creates an index
/// </summary>
/// <exception cref="System.ArgumentException">Required parameters not provided or invalid.</exception>
```

Reply:
Thanks for your feedback @DiscoPYF.
Resolved!